x-pack/auditbeat/module/system/user/user.go

```diff
@@ -288,44 +289,55 @@ func (ms *MetricSet) Fetch(report mb.ReporterV2) {
 
 // reportState reports all existing users on the system.
 func (ms *MetricSet) reportState(report mb.ReporterV2) error {
+	var errs multierror.Errors
 	ms.lastState = time.Now()
 
 	users, err := GetUsers(ms.config.DetectPasswordChanges)
 	if err != nil {
-		return errors.Wrap(err, "failed to get users")
+		errs = append(errs, errors.Wrap(err, "error while getting users"))
 	}
+
 	ms.log.Debugf("Found %v users", len(users))
+	if len(users) > 0 {
+		stateID, err := uuid.NewV4()
+		if err != nil {
+			errs = append(errs, errors.Wrap(err, "error generating state ID"))
+		}
 
-	stateID, err := uuid.NewV4()
-	if err != nil {
-		return errors.Wrap(err, "error generating state ID")
-	}
-	for _, user := range users {
-		event := ms.userEvent(user, eventTypeState, eventActionExistingUser)
-		event.RootFields.Put("event.id", stateID.String())
-		report.Event(event)
-	}
+		for _, user := range users {
+			event := ms.userEvent(user, eventTypeState, eventActionExistingUser)
+			event.RootFields.Put("event.id", stateID.String())
```

Comment:
stateID will be an all zero uuid if  `uuid.NewV4()` failed. What's the impact of this on the metricset and the consumers of this data?

Reply:
When sending a list of all users, the `event.id` is what they have in common, to make it easy for consumers to correlate them. If generating it failed, it would no longer be possible to use it for that purpose - lists of users sent from the same system at different times would have the same (zero) event.id. Instead, they would have to fall back on the less user-friendly timestamp.

It's unlikely to happen, but if it did the dataset would still send all the users, and an error as well.